skcosmo/linear_model/_base.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+from scipy.linalg import orthogonal_procrustes
+
+from sklearn.base import RegressorMixin, MultiOutputMixin
+from sklearn.linear_model import LinearRegression
+
+
+class OrthogonalRegression(MultiOutputMixin, RegressorMixin):
+    """Orthogonal regression by solving the Procrustes problem
+
+    Parameters
+    ----------
+    project_to_prediction_space : bool, default=False
+        Controls if X is projected to the feature space of y.
+        If it is set to False X and y are padded with zeros to the larger
+        number of features of X  and y. The projection method is similar
+        to the procedure in the computation GFRD in the first version of
+        Ref. [frm]_. The method has been adapted to project to the y feature
+        space. The original version projected always to the lower dimensional
+        feature space of X and y. This can introduce nonanalytic behavior with
+        respect to changes in dimensions of X for cases
+        where X n_features > y n_features.
+
+        See ``examples/linear_model/plot_orthogonal_regression_nonanalytic_behavior.py``
+
+    linear_estimator : object implementing fit/predict, default=None
+        The linear estimator is used when `project_to_prediction_space`
+        is set to True, to compute the projection matrix
+
+    References
+    ----------
+    [frm]_ Goscinski, Fraux, and Ceriotti. "The role of feature space in
+           atomistic learning." arXiv preprint arXiv:2009.02741 (2020).
+    """
+
+    def __init__(self, project_to_prediction_space=True, linear_estimator=None):
+        self.project_to_prediction_space = project_to_prediction_space
+        self.linear_estimator = linear_estimator
+
+    def fit(self, X, y):
+        """
+        Parameters
+        ----------
+        X : array_like of shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array_like of shape (n_samples, n_targets)
+            Training data, where n_samples is the number of samples
+            and n_targets is the number of target properties.
+        """
+        self.n_components = max(X.shape[1], y.shape[1])
+        X = np.pad(X, [(0, 0), (0, self.n_components - X.shape[1])])
+
+        if self.project_to_prediction_space:
+            # check estimator
+            linear_estimator = (
+                LinearRegression()
+                if self.linear_estimator is None
+                else self.linear_estimator
+            )
+            # compute projection matrix
+            linear_estimator.fit(X, y)
+            # try scipy sparse return only u
+            U, _, _ = np.linalg.svd(linear_estimator.coef_.T, full_matrices=False)
+            X = X.dot(U)
```

Comment:
sklearn uses dot

Reply:
Given that `numpy` is moving away from `numpy.dot` for matrices, it is likely that `sklearn` will as well. We should adhere to the newer standard rather than have to redo things if `sklearn` does. Besides which, it is more specific to use the matrix-specific functions rather than `dot`, which can read ambigiously.

Reply:
okay